app/models/agreement_letter.rb

```diff
@@ -90,4 +95,13 @@ def too_big?(file)
     def wrong_filetype?(file)
       file.content_type != ALLOWED_MIMETYPE
     end
+
+  def wrong_file_trailer? file
+    begin
+    page_analysis = PDF::Inspector::Page.analyze_file(file.open)
+    false
+    rescue
```

Comment:
Please only rescue `PDF::Reader::UnsupportedFeatureError` and `PDF::Reader::MalformedPDFError` to avoid hiding other exceptions.